### PR TITLE
Show constitution axioms and reality in the evaluation report

### PR DESCRIPTION
**Pull request by isabelcabezasm**
This pull request implements a multi-phase plan to enhance the evaluation report system by displaying constitutional axioms and reality items with full descriptions, improving UI/UX, and ensuring robust test coverage and documentation. The changes span Python data models, evaluation pipeline logic, HTML/CSS/JavaScript front-end, and comprehensive testing. The most important changes are grouped below:

**Evaluation Report UI Enhancements:**

- Added collapsible, responsive sections to the HTML report for axiom and reality definitions, with each item’s ID and description shown in a styled card layout. Reference tags in evaluation results now display tooltips with full descriptions on hover, improving clarity and usability.

**Backend Data Model and Pipeline Updates:**

- Extended Python data models (`AxiomItem`, `RealityItem`, and `EvaluationResult`) to store axiom and reality definitions. The evaluation pipeline now loads these definitions from JSON files and includes them in the output, maintaining backward compatibility.

**Testing and Edge Case Handling:**

- Added and updated over 60 JavaScript and 40 Python tests to cover new features, edge cases (such as missing or malformed definitions), collapsible UI elements, and integration of definitions into report generation. Ensured all tests and lint checks pass.

**Documentation:**

- Updated `README.md` with a new section detailing report features, including collapsible axiom/reality sections, tooltips, and responsive design. Verified and completed in-code documentation (JSDoc and Python docstrings) for all new and modified functions and models. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R143-R178) [[2]](diffhunk://#diff-04afe43537bc3179243f0a9921f740397c8cdb50cec6d72ba66f1e23026957acR1-R381)

### CODE REVIEW

File: src/eval/report_generation/templates/script.js [hunk 1]
```diff
@@ -662,6 +728,112 @@ function toggleEvaluation(headerElement) {
...
+    const definitionsHtml = definitions
+        .map(item => `
+            <div class="definition-item">
+                <span class="definition-id">${item.id}</span>
+                <span class="definition-text">${item.description}</span>
+            </div>
+        `)
+        .join('');
+
+    container.innerHTML = definitionsHtml;
```
[the remark this answers is not in the document]
Comment (isabelcabezasm) [reply src/eval/report_generation/templates/script.js]: 18fbbfa2223e298fdaea82ec1b5da157e71ac815

File: src/eval/report_generation/templates/script.js [hunk 2]
```diff
@@ -662,6 +728,112 @@ function toggleEvaluation(headerElement) {
...
```
[the remark this answers is not in the document]
Comment (isabelcabezasm) [reply src/eval/report_generation/templates/script.js]: 18fbbfa2223e298fdaea82ec1b5da157e71ac815

File: src/eval/report_generation/templates/script.js [hunk 3]
```diff
@@ -219,18 +233,63 @@ function highlightEntitiesInText(text, entities) {
...
+        // Escape HTML entities in description for safe attribute value
+        const escapedDescription = description
+            .replace(/&/g, '&amp;')
+            .replace(/"/g, '&quot;')
+            .replace(/</g, '&lt;')
+            .replace(/>/g, '&gt;');
+        return `<span class="reference-tag ${tagClass}" data-tooltip="${escapedDescription}">${refId}</span>`;
+    }
+    return `<span class="reference-tag ${tagClass}">${refId}</span>`;
+}
```
[the remark this answers is not in the document]
Comment (isabelcabezasm) [reply src/eval/report_generation/templates/script.js:294]: 18fbbfa2223e298fdaea82ec1b5da157e71ac815

File: tests/ui/report-script.spec.ts [hunk 4]
```diff
@@ -0,0 +1,824 @@
...
+        { id: "A-001", description: "Description with <script>alert('xss')</script>" }
+      ]
+    };
+
+    renderAxiomDefinitions();
+
+    const container = document.getElementById("axiom-definitions-list");
+    // The description is rendered as-is (DOM escaping happens at render)
+    expect(container?.querySelector(".definition-item")).not.toBeNull();
+  });
```
[the remark this answers is not in the document]
Comment (isabelcabezasm) [reply tests/ui/report-script.spec.ts:810]: e22630dad188e305b382af3399443cb55db72ef2

File: README.md [hunk 5]
```diff
@@ -140,6 +140,42 @@ The system automatically handles this - just ensure:
...
+  on mobile devices
+- **Collapsible Sections**: Definition sections can be collapsed/expanded to manage visual space
+
+### Report Data
+
+The report reads evaluation data from `evaluation_results.json`, which includes:
+
+- `axiom_definitions`: Array of axiom items with `id` and `description` fields
+- `reality_definitions`: Array of reality items with `id` and `description` fields
+- `evaluations`: Array of evaluation results with expected/found references and verdicts
```
[the remark this answers is not in the document]
Comment (isabelcabezasm) [reply README.md]: 6c7f8ee324a5f6f99521766e0176110cd590c19e

File: src/eval/report_generation/templates/script.js [hunk 6]
```diff
@@ -662,6 +728,112 @@ function toggleEvaluation(headerElement) {
...
+            '<p class="no-definitions">No axiom definitions available.</p>';
+        return;
+    }
+
+    const definitionsHtml = definitions
+        .map(item => `
+            <div class="definition-item">
+                <span class="definition-id">${item.id}</span>
+                <span class="definition-text">${item.description}</span>
+            </div>
```
[the remark this answers is not in the document]
Comment (isabelcabezasm) [reply src/eval/report_generation/templates/script.js]: The XSS vulnerabilities have already been fixed. The code now uses two secure approaches:

renderAxiomDefinitions: Uses the [escapeHtml()](vscode-file://vscode-app/c:/Users/isacabe/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) helper function with template literals
renderRealityDefinitions: Uses DOM manipulation with [textContent](vscode-file://vscode-app/c:/Users/isacabe/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html), which automatically prevents XSS
Both approaches are secure and follow defense-in-depth principles.

File: src/eval/report_generation/templates/script.js [hunk 7]
```diff
@@ -219,18 +233,63 @@ function highlightEntitiesInText(text, entities) {
...
+function renderReferenceTag(refId, tagClass, definitionsMap) {
+    const description = definitionsMap.get(refId);
+    if (description) {
+        // Escape HTML entities in description for safe attribute value
+        const escapedDescription = description
+            .replace(/&/g, '&amp;')
+            .replace(/"/g, '&quot;')
+            .replace(/</g, '&lt;')
+            .replace(/>/g, '&gt;');
+        return `<span class="reference-tag ${tagClass}" data-tooltip="${escapedDescription}">${refId}</span>`;
```
[the remark this answers is not in the document]
Comment (isabelcabezasm) [reply src/eval/report_generation/templates/script.js]: 4177902aca42c426e804e8d8542bcd3f1ab1df1b

File: src/eval/report_generation/templates/styles.css [hunk 8]
```diff
@@ -350,6 +350,78 @@ body {
...
+/* Show tooltip on focus for accessibility */
+.reference-tag[data-tooltip]:focus::after,
+.reference-tag[data-tooltip]:focus::before {
+    opacity: 1;
+    visibility: visible;
+}
+
+/* Allow focus on reference tags with tooltips */
+.reference-tag[data-tooltip] {
+    outline: none;
```
[the remark this answers is not in the document]
Comment (isabelcabezasm) [reply src/eval/report_generation/templates/styles.css]: 7b5af161194a72c9181a50cd8b4340717a4d04ef

File: src/eval/report_generation/templates/script.js [hunk 9]
```diff
@@ -219,18 +233,63 @@ function highlightEntitiesInText(text, entities) {
...
+    }
+    return `<span class="reference-tag ${tagClass}">${refId}</span>`;
```
[the remark this answers is not in the document]
Comment (isabelcabezasm) [reply src/eval/report_generation/templates/script.js]: The template file at script.js:285-294 already has the proper XSS protection. 
Both refId and description are escaped using escapeHtml() on lines 286 and 290, and escapedRefId is used in the return statements on lines 291 and 293.

File: src/eval/report_generation/templates/script.js [hunk 10]
```diff
@@ -662,6 +746,122 @@ function toggleEvaluation(headerElement) {
...
+        const textSpan = document.createElement('span');
+        textSpan.className = 'definition-text';
+        textSpan.textContent = item.description;
+
+        itemDiv.appendChild(idSpan);
+        itemDiv.appendChild(textSpan);
+
+        container.appendChild(itemDiv);
+    });
+}
```
[the remark this answers is not in the document]
Comment (isabelcabezasm) [reply src/eval/report_generation/templates/script.js:879]: c7b45f241d763fa72b93b111ec53521a91899d29

File: tests/ui/report-script.spec.ts [hunk 11]
```diff
@@ -0,0 +1,848 @@
...
+      reality_definitions: [{ id: "R-001", description: longDescription }]
+    };
+
+    renderRealityDefinitions();
+
+    const container = document.getElementById("reality-definitions-list");
+    const textElement = container?.querySelector(".definition-text");
+    expect(textElement?.textContent).toBe(longDescription);
+  });
+});
```
[the remark this answers is not in the document]
Comment (isabelcabezasm) [reply tests/ui/report-script.spec.ts:866]: a0b8d7e7d5a3669a97a088a8f630d0491a6e5ad6

File: src/eval/report_generation/templates/index.html [hunk 12]
```diff
@@ -56,6 +56,40 @@ <h3 id="overall-score">0.00</h3>
...
+                    </p>
+                    <div id="axiom-definitions-list" class="definitions-list">
+                        <!-- Axiom definitions will be populated by JavaScript -->
+                    </div>
+                </div>
+            </div>
+
+            <div class="definitions-section reality-definitions">
+                <div class="definitions-header" role="button" tabindex="0" aria-expanded="true"
+                    onclick="toggleDefinitionsSection(this)">
```
[the remark this answers is not in the document]
Comment (isabelcabezasm) [reply src/eval/report_generation/templates/index.html]: 04c76140e8f27989d139de76d02960f5b494df12
